### ISSUE TITLE
📷 QR Scanner in Migration

### ISSUE THREAD

**Issue by Abby-Wheelis** [e-mission/e-mission-docs#982]
As we're migrating the onboarding [👋 New Onboarding Flow + Switch to React Navigation](https://github.com/e-mission/e-mission-phone/pull/1032#top) we need a new library for scanning the QR code (opcode) on the join page, as we previously used an angular library. 

The tricky part here is that the app is not fully React-Native, it is using react-native-web, so we need a library that works with our current configuration. I've started doing some research into qr code libraries, and wanted to document some of my findings.

- [`react-qr-reader`](https://www.npmjs.com/package/react-qr-reader) is designed for React webapps, and works [I tested with this example I found](https://stackblitz.com/edit/react-qr-reader?file=src%2FApp.js), but the last release was in 2019 and last update to the code base in March 2022, so it does not seem to be extensively maintained. It does seem to work based on the example I found, and seems easy to use.
- [`react-qr-scanner`](https://github.com/yudielcurbelo/react-qr-scanner) worked when I tried the [demo](https://yudielcurbelo.github.io/react-qr-scanner/?path=/story/media-device--list-devices), the documentations makes it seem easy to use, and the last updates were July of this year.

There are lots of libraries out there, I think it's just a matter of finding the one we feel most comfortable with it staying maintained.

**Comment by Abby-Wheelis** [comment]
This morning, I took a moment to look into what we're currently using, which seems to be [`phonegap-plugin-barcodescanner`](https://github.com/phonegap/phonegap-plugin-barcodescanner), which is compatible with phonegap and cordova. However, this repo has been read-only since August 2020, and seems to be deprecated. We might have been able to keep using it since it operates over cordova and not necessarily angular directly, but I think `react-qr-scanner` (if it works in our current configuration) would be a better choice, and we might as well swap sooner rather than later.

**Comment by JGreenlee** [comment]
Note that the replacement library doesn't necessarily **have** to be a React implementation.
If we can find a lightweight, framework-independent QR scanning library, that should work just as well (and potentially even be preferred)

If `react-qr-scanner` looks good, no issue with using it-- but if not, there do seem to be other options.

**Comment by shankari** [comment]
I have looked into replacements for `phonegap-plugin-barcodescanner` before, but the other plugins seemed really sketchy, and didn't actually work. Which is why I have not touched this at all. If you search for closed issues with `phonegap-plugin-barcodescanner` and focus on previous upgrade issues, you should find my notes and my thoughts around why they seemed sketchy.

@Abby-Wheelis how did you test this? Neither emulator typically supports the camera, so I am not sure how you would know that it was scanning properly.

QR scanner is a popular feature so lots of people try to write one. Please double-check to make sure that the source seems legit and that the code doesn't have any other weird stuff in it.

**Comment by Abby-Wheelis** [comment]
With that in mind, I'm working on trialing [qr-scanner](https://www.npmjs.com/package/qr-scanner?activeTab=readme). I've installed and started trying to work with it in the join page. No breaking errors so far - built with the new package but I'll have to wait and test when I'm off the lab's network tomorrow. In the emulator it's behaving as if the camera is detected, but isn't actually using the webcam.

**Comment by Abby-Wheelis** [comment]
>how did you test this? Neither emulator typically supports the camera, so I am not sure how you would know that it was scanning properly.

I had found demos of those individual implementations, and was using the online demos as a proof of concept. 

>QR scanner is a popular feature so lots of people try to write one. Please double-check to make sure that the source seems legit and that the code doesn't have any other weird stuff in it.
>>BarcodeScanner currently archived (https://github.com/phonegap/phonegap-plugin-barcodescanner). No suggested replacement. Current version on GitHub is 8.1.0. Current version on npm is 8.1.2 and is published by "@modjoulteam". They claim to have their own repo (https://github.com/Modjoul-Team/phonegap-plugin-barcodescanner), but it doesn't really exist. Switching to installing from GitHub to avoid hijacking. May have to fork and upgrade instead.

The one that I found and am trying to work with does not seem sketchy, but I don't know if it works yet (going to try using the devapp on my test phone). The documentation is solid, it seems to be highly downloaded, and is open source on github. I don't have much experience choosing libraries though, so definitely open to other opinions or options, you can look at `qr-scanner` [here](https://www.npmjs.com/package/qr-scanner) or [on github](https://github.com/nimiq/qr-scanner).

**Comment by Abby-Wheelis** [comment]
I tried for a while last night and never got the new library to work -- it needs a reference to an html `<video>` element to operate, and I have successfully embedded the element, but can't seem to get a reference to it using anything like `getElementById`. I can try more tonight.

I did also try working with the older cordova plugin, which was still seamlessly able to pull up a camera scanning window, so we could keep using that for now if we wanted.
